pkg/util/containers/v2/metrics/collector_ecs_fargate_test.go

```diff
@@ -0,0 +1,291 @@
+					"eth1": {BytesSent: floatPtr(2398415937), PacketsSent: floatPtr(1898631), BytesRcvd: floatPtr(1259037719), PacketsRcvd: floatPtr(428002)},
+				},
+				BytesRcvd:   floatPtr(3657453656),
+				PacketsRcvd: floatPtr(2326633),
+				BytesSent:   floatPtr(3657453656),
+				PacketsSent: floatPtr(2326633),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert.EqualValues(t, tt.want, convertEcsNetworkStats(tt.args.netStats, tt.args.networks))
+		})
+	}
+}
+
+func TestConvertEcsStats(t *testing.T) {
+	constTime := time.Now()
+	type args struct {
+		ecsStats *v2.ContainerStats
+	}
+	tests := []struct {
+		name string
+		args args
+		want *ContainerStats
+	}{
+		{
+			name: "nominal case",
+			args: args{
+				ecsStats: &v2.ContainerStats{
+					CPU:    v2.CPUStats{Usage: v2.CPUUsage{Total: 1137691504, Kernelmode: 80000000, Usermode: 810000000}},
+					Memory: v2.MemStats{Limit: 9223372036854772000, Usage: 6504448, Details: v2.DetailedMem{RSS: 4669440, Cache: 651264}},
+					IO: v2.IOStats{BytesPerDeviceAndKind: []v2.OPStat{
+						{
+							Major: 202,
+							Minor: 26368,
+							Kind:  "Read",
+							Value: 638976,
+						},
+						{
+							Major: 202,
+							Minor: 26368,
+							Kind:  "Write",
+							Value: 0,
+						},
+						{
+							Major: 202,
+							Minor: 26368,
+							Kind:  "Sync",
+							Value: 638976,
+						},
+						{
+							Major: 202,
+							Minor: 26368,
+							Kind:  "Async",
+							Value: 0,
+						},
+						{
+							Major: 202,
+							Minor: 26368,
+							Kind:  "Total",
+							Value: 638976,
+						},
+					},
+						OPPerDeviceAndKind: []v2.OPStat{
+							{
+								Major: 202,
+								Minor: 26368,
+								Kind:  "Read",
+								Value: 12,
+							},
+							{
+								Major: 202,
+								Minor: 26368,
+								Kind:  "Write",
+								Value: 0,
+							},
+							{
+								Major: 202,
+								Minor: 26368,
+								Kind:  "Sync",
+								Value: 12,
+							},
+							{
+								Major: 202,
+								Minor: 26368,
+								Kind:  "Async",
+								Value: 0,
+							},
+							{
+								Major: 202,
+								Minor: 26368,
+								Kind:  "Total",
+								Value: 12,
+							},
+						},
+					},
+				},
+			},
+			want: &ContainerStats{
+				Timestamp: constTime,
+				CPU:       &ContainerCPUStats{Total: floatPtr(1137691504), System: floatPtr(80000000), User: floatPtr(810000000)},
+				Memory:    &ContainerMemStats{Limit: floatPtr(9223372036854772000), UsageTotal: floatPtr(6504448), RSS: floatPtr(4669440), Cache: floatPtr(651264)},
+				IO:        &ContainerIOStats{ReadBytes: floatPtr(638976), WriteBytes: floatPtr(0), ReadOperations: floatPtr(12), WriteOperations: floatPtr(0)},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := convertEcsStats(tt.args.ecsStats)
+			got.Timestamp = constTime
```

Comment:
Nit: not sure if it's worth adding a comment to clarify that this is to avoid checking the timestamp because we have no control over it without as the code is right now.